src/main/java/org/folio/rest/impl/RelatedInstanceAPI.java

```diff
@@ -0,0 +1,100 @@
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+/**
+ * Implements the related instance persistency using postgres jsonb.
+ */
+public class RelatedInstanceAPI implements org.folio.rest.jaxrs.resource.InstanceStorageRelatedInstances {
+  public static final String RELATED_INSTANCE_TABLE = "related_instance";
+
+  @Override
+  public void getInstanceStorageRelatedInstances(String query, int offset, int limit, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.get(RELATED_INSTANCE_TABLE, RelatedInstance.class, RelatedInstances.class, query, offset, limit,
+      okapiHeaders, vertxContext, GetInstanceStorageRelatedInstancesResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void postInstanceStorageRelatedInstances(String lang, RelatedInstance entity, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    if (checkInstanceRelatedToSelf(entity)) {
+      asyncResultHandler.handle(cannotBeRelatedToSelf());
+      return;
+    }
+
+    new RelatedInstanceService(vertxContext, okapiHeaders)
+      .createRelatedInstance(entity)
+      .onSuccess(response -> asyncResultHandler.handle(Future.succeededFuture(response)))
+      .onFailure(handleFailure(asyncResultHandler));
+  }
+
+  @Override
+  public void getInstanceStorageRelatedInstancesByRelatedInstanceId(String instanceId, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.getById(RELATED_INSTANCE_TABLE, RelatedInstance.class, instanceId,
+        okapiHeaders, vertxContext, GetInstanceStorageRelatedInstancesByRelatedInstanceIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void deleteInstanceStorageRelatedInstancesByRelatedInstanceId(String instanceId, String lang,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+
+    PgUtil.deleteById(RELATED_INSTANCE_TABLE, instanceId, okapiHeaders, vertxContext,
+      DeleteInstanceStorageRelatedInstancesByRelatedInstanceIdResponse.class, asyncResultHandler);
+  }
+
+  @Override
+  public void putInstanceStorageRelatedInstancesByRelatedInstanceId(String instanceId, String lang, RelatedInstance entity,
+      Map<String, String> okapiHeaders, Handler<AsyncResult<Response>> asyncResultHandler,
+      Context vertxContext) {
+    if (checkInstanceRelatedToSelf(entity)) {
+        asyncResultHandler.handle(cannotBeRelatedToSelf());
+        return;
+    }
+    new RelatedInstanceService(vertxContext, okapiHeaders)
+      .updateRelatedInstance(entity)
+      .onSuccess(response -> asyncResultHandler.handle(Future.succeededFuture(response)))
+      .onFailure(handleFailure(asyncResultHandler));
+  }
```

Comment:
This seems to take the instance ID as a parameter. Does that mean that there can only be a single related instance for each instance? If not, how does the client refer to each related instance record?

Reply:
See above.  Relationships are modified and deleted by their own unique UUID, not the UUID of the instance records involved.